src/test/scala/strawman/collection/test/BuildFromTest.scala

```diff
@@ -0,0 +1,116 @@
+package strawman
+package collection.test
+
+import org.junit.Test
+import strawman.collection.Iterable
+import strawman.collection.mutable.Builder
+import strawman.collection._
+
+import scala.{Any, Boolean, Either, Int, Left, None, Option, PartialFunction, Right, Some, Unit}
+import scala.Predef.{ArrowAssoc, implicitly}
+import scala.math.Ordering
+import java.lang.String
+
+class BuildFromTest {
```

Comment:
I really like the usage here.  There's a bit of confusing magic about "how does it even know the collection type I mean when I didn't really say what it was?"

But the magic works and it feels way more usable to me than CBF did. Especially since `xs` is now just a plain `Iterable[Option[A]]` instead of some generified parameterized thingy that might be hard to get right.